### PR TITLE
stale element exception when getText

### PR DESCRIPTION
**Pull request by aweng98**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by aweng98** [comment]
`submissionDetailsPage.abortSubmission()` needs call to `awaitReady()` after click Abort button.

```
[info] MethodConfigSpec:
[info] - abort a submission *** FAILED *** (22 seconds, 430 milliseconds)
[info]   org.openqa.selenium.StaleElementReferenceException: stale element reference: element is not attached to the page document
...
[info]   at org.openqa.selenium.remote.RemoteWebElement.getText(RemoteWebElement.java:160)
[info]   at org.scalatest.selenium.WebBrowser$Element$class.text(WebBrowser.scala:1304)
[info]   at org.scalatest.selenium.WebBrowser$$anon$1.text(WebBrowser.scala:3179)
[info]   at org.broadinstitute.dsde.firecloud.FireCloudView$$anonfun$readText$1.apply(FireCloudView.scala:14)
```

**Review by rushtong — APPROVED** [review]
👍 LGTM

**Review by MattPutnam — CHANGES_REQUESTED** [review]
I think this is waiting on the wrong thing. We should wait on the abort confirmation modal to disappear.

Instead of having `submissionAbortModalConfirmationButton` be a member of `SubmissionDetailsPage`, the `submissionAbortButton.doClick()` should spawn a `Modal` that we then `submit()`.

**Comment by aweng98** [comment]
that's a good solution. done. ty.

**Comment by aweng98** [comment]
hold and retest until https://github.com/broadinstitute/firecloud-ui/pull/1239/ is merged.

**Comment by aweng98** [comment]
Matt, your suggestion is very good. I refactored code. About the UI code changes, I think it's not needed in this case.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala [hunk 1]
```diff
@@ -77,8 +76,34 @@ class SubmissionDetailsPage(namespace: String, name: String, var submissionId: S
...
+
+  def abortSubmission(): Unit = {
+    val confirmModal = clickAbortButton()
+    confirmModal.submit()
+  }
+
+  /**
+    * page class for Submission Abort Confirmation modal
+    */
+  class AbortSubmissionConfirmationModal(implicit webDriver: WebDriver) extends ConfirmModal {
```
Comment (MattPutnam) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala:92]: We shouldn't need this class. That's the main reason for having the confirm-modal distinction in the UI.

Comment (aweng98) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala:92]: hmm, it is to verify expected text displayed in expected Confirm modal. Displayed texts on different Confirm modals are going to be different.

Comment (MattPutnam) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala:92]: I feel like we don't need to test that. We should be testing broad behavior, not wording choice. At a high level, we click the "abort" button, and we click "confirm" on the modal that pops up.

Comment (aweng98) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala:92]: I agree, to a certain degree. one last concern (question), if text is not visible somehow on UI in this modal, do you think test should still pass? without this check, test would pass. 
I am concerned test would pass when it should have failed for some unexpected reasons.

Comment (aweng98) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala:92]: Hi Matt, I tested this to be sure it works. What are your thoughts on this? Thank you for your attention.

Comment (MattPutnam) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala:92]: Again, I don't want to subclass ConfirmModal. It's not designed to be subclassed.

Comment (aweng98) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala:92]: In that case, can you go ahead merge your PR https://github.com/broadinstitute/firecloud-ui/pull/1239? You have a good solution.

**Pull request closed without merging**